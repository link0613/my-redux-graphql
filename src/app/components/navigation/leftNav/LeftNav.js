// @flow

// #region imports
import React from 'react';
import LeftNavButton from './leftNavButton/LeftNavButton';
// #endregion

// #region flow types
export type NavItem = {
  link: string,
  label: string,
  viewName: string,
};

export type Props = {
  leftLinks: Array<NavItem>,
  onLeftNavButtonClick: () => {},

  ...any,
};
// #endregion

const LeftNav = ({ leftLinks, onLeftNavButtonClick }: Props) => (
  <ul className="nav navbar-nav">
    {leftLinks.map(({ link, label, viewName }, index) => {
      return (
        <LeftNavButton
          key={index}
          link={link}
          label={label}
          viewName={viewName}
          onClick={onLeftNavButtonClick}
        />
      );
    })}
  </ul>
);

// #region static props
LeftNav.displayName = 'LeftNav';
// #endregion

export default LeftNav;
