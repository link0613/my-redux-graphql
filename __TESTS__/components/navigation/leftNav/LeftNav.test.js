import React from 'react';
import LeftNav from '../../../../src/app/components/navigation/leftNav/LeftNav';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';

describe('LeftNav (navigation) component', () => {
  it('renders as expected', () => {
    const props = {
      onLeftNavButtonClick: () => {},
      leftLinks: [
        {
          link: 'link',
          label: 'label',
          viewName: '/',
        },
      ],
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <LeftNav {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
