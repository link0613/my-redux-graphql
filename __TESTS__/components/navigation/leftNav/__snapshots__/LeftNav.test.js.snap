// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNav (navigation) component renders as expected 1`] = `
<div>
  <ul
    className="nav navbar-nav"
  >
    <li>
      <a
        href="link"
        onClick={[Function]}
      >
        label
      </a>
    </li>
  </ul>
</div>
`;
